aurellion/src/dapp-connectors/dapp-controller.ts

```diff
@@ -194,3 +201,45 @@ export const assignDriverToJobId = async (jobID: string) => {
     console.error("Error in assignDriverToJobId:", error);
   }
 };
+
+export const fetchDriverUnassignedJourneys = async () => {
+  const jobIds = [];
+  const journeys: Journey[] = [];
+  try {
+    const signer = await getSigner();
+    if (!signer) {
+      throw new Error("Signer is undefined");
+    }
+    const contract = new ethers.Contract(
+      REACT_APP_AUSYS_CONTRACT_ADDRESS,
+      contractABI,
+      signer
+    );
+    const totalJobsCount = await contract.jobIdCounter();
+    // Index starts with 1 because smart contract jobIdCounter starts at 1
+    for (let i = 1; i <= totalJobsCount; i++) {
+      try {
+        const jobId = await contract.numberToJobID(i);
+        jobIds.push(jobId);
+      } catch (error) {
+        console.error("Error retrieving jobId", error);

```

Comment:
does console.error automatically throw? if so this could cause the app to stop when we dont want it to. I need to be more familiar with console.error but since it is not neccesarily an error and doing it this way provides better context to whats going on than a generic error sstatement it may be better to simply:
```suggestion
        console.log('No job exists at index ${i}')
```

Reply:
Agree this would be better. The difference between .log and .error is to which stream it gets printed to. Looks like in react native, they both print to the standard terminal either way